streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -640,6 +644,78 @@ public void shouldRespectCommitNeeded() {
         assertFalse(task.commitNeeded());
     }
 
+    @Test
+    public void shouldRestorePartitionTimeAfterRestartWithEosDisabled() {
+        task = createStatelessTask(createConfig(false));
+        task.initializeStateStores();
+        task.initializeTopology();
+
+        task.addRecords(partition1, singletonList(getConsumerRecord(partition1, DEFAULT_TIMESTAMP)));
+
+        task.process();
+
+        task.commit();
+        assertTrue(task.decodeTimestamp(consumer.committed(partition1).metadata())
+                   == DEFAULT_TIMESTAMP);
+        // reset times here to artificially represent a restart
+        task.resetTimes();
+        assertTrue(task.partitionTime(partition1) == RecordQueue.UNKNOWN);
+
+        task.initializeTaskTime();
+        assertTrue(task.partitionTime(partition1) == DEFAULT_TIMESTAMP);
+        assertTrue(task.streamTime() == DEFAULT_TIMESTAMP);
+    }
+
+    @Test
+    public void shouldRestorePartitionTimeAfterRestartWithEosEnabled() {
+        task = createStatelessTask(createConfig(true));
+        task.initializeStateStores();
+        task.initializeTopology();
+
+        task.addRecords(partition1, singletonList(getConsumerRecord(partition1, DEFAULT_TIMESTAMP)));
+
+        task.process();
+        task.commit();
+
+        //since consumer is mock, there is no real broker
+        //so we need to manually commit the information to stimulate broker
+        final Map<TopicPartition, OffsetAndMetadata> offsetMap = new HashMap<>();
+        final String encryptedMetadata = task.encodeTimestamp(DEFAULT_TIMESTAMP);
+        offsetMap.put(partition1, new OffsetAndMetadata(DEFAULT_TIMESTAMP, encryptedMetadata));
+        consumer.commitSync(offsetMap);
+        assertTrue(task.decodeTimestamp(consumer.committed(partition1).metadata())
+                   == DEFAULT_TIMESTAMP);
+
+        long partitionTime = -1L;
+        final List<Map<String, Map<TopicPartition, OffsetAndMetadata>>> metadataList = 
+            producer.consumerGroupOffsetsHistory();
+        for (final Map<String, Map<TopicPartition, OffsetAndMetadata>> map : metadataList) {
```

Comment:
No need to use a nested for loops. This can be simplified to:
``` 
final String metadata = metadataList.get(topic1).get(partition1).metadata();
assertThat(task.decodeTimestamp(metadata), equalTo(DEFAULT_TIMESTAMP));
```